### ISSUE TITLE
Test: NullPointer Exception for `org.logstash.ackedqueue.Queue.getPersistedByteSize`

### ISSUE THREAD

**Issue by ph** [elastic/logstash#7022]
https://logstash-ci.elastic.co/job/elastic+logstash+5.x+multijob-os-compatibility/os=debian/264/console

```
  1) LogStash::Api::Modules::Plugins should return a 404 to unknown request
     Failure/Error: Unable to find matching line from backtrace
     Java::JavaLang::NullPointerException:
     # org.logstash.ackedqueue.Queue.getPersistedByteSize(Queue.java:143)
     # org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue.ruby_persisted_size_in_bytes(JrubyAckedQueueExtLibrary.java:117)
     # org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$0$0$ruby_persisted_size_in_bytes.call(JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$0$0$ruby_persisted_size_in_bytes.gen)
     # org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
     # org.jruby.ast.CallNoArgNode.interpret(CallNoArgNode.java:60)
     # org.jruby.ast.CallTwoArgNode.interpret(CallTwoArgNode.java:59)
     # org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
```

```
rspec /var/lib/jenkins/workspace/elastic+logstash+5.x+multijob-os-compatibility/os/debian/logstash-core/spec/support/shared_examples.rb:99 # LogStash::Api::Modules::Plugins should return a 404 to unknown request
rspec /var/lib/jenkins/workspace/elastic+logstash+5.x+multijob-os-compatibility/os/debian/logstash-core/spec/api/lib/api/node_plugins_spec.rb:29 # LogStash::Api::Modules::Plugins retrieving plugins should return the total number of plugins

Randomized with seed 23098
```

**Comment by ph** [comment]
This seems to be coming from https://github.com/elastic/logstash/commit/49993c5220dea8a1b219e4026d03fae1fd6c235a

**Comment by original-brownbear** [comment]
@colinsurprenant @suyograo tests for https://github.com/elastic/logstash/pull/7010 incoming here

**Comment by original-brownbear** [comment]
just for the record, this can be reproduced via:

```java
    public void getsPersistedByteSizeCorrectly() throws Exception {
        Settings settings = TestSettings.persistedQueueSettings(100, dataPath);
        try (Queue q = new Queue(settings)) {
            final Queueable elem = new StringElement("fooo");
            for (int i = 0; i < 100; ++i) {
                q.write(elem);
            }
            q.open();
            assertThat(q.getPersistedByteSize(), is(equalTo(1L)));
        }
    }
}
```

Writing to a queue before calling `open` on it gets it in a state where wee get an NPE from the `getPersistedBytes`.

**Comment by original-brownbear** [comment]
ok not that many tests added but fix is in #7023

**Comment by original-brownbear** [comment]
@ph should be fixed by #7023 :)
